SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,142 @@ final public class Repository {
+				}
+				git_tree_free(unsafeParentTree)
+
+				var unsafeDiff: OpaquePointer? = nil
+				let diffResult = git_diff_tree_to_tree(&unsafeDiff, self.pointer, unwrapParentTree, unwrapBaseTree, nil)
+				guard diffResult == GIT_OK.rawValue, let unwrapDiffResult = unsafeDiff else {
+					return Result.failure(NSError(gitError: diffResult, pointOfFailure: "git_diff_tree_to_tree"))
+				}
+
+				if mergeDiff == nil {
+					mergeDiff = unwrapDiffResult
+				} else {
+					let mergeResult = git_diff_merge(mergeDiff, unwrapDiffResult)
+					guard mergeResult == GIT_OK.rawValue else {
+						return Result.failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
+					}
+				}
+			}
+			return self.processDiffDeltas(mergeDiff!)
+		}
+	}
+
+	private func processDiffDeltas(_ diffResult: OpaquePointer) -> Result<[DiffDelta], NSError> {
+		var returnDict = [DiffDelta]()
+
+		let count = git_diff_num_deltas(diffResult)
+
+		for i in 0..<count {
+			let delta = git_diff_get_delta(diffResult, i)
+			let gitDiffDelta = DiffDelta(from: (delta?.pointee)!)
+
+			returnDict.append(gitDiffDelta)
+
+			git_diff_free(OpaquePointer(delta))
+		}
+
+		let result = Result<[DiffDelta], NSError>.success(returnDict)
+		return result
+	}
+
+	// MARK: - Status
+
+	public func status() -> Result<[StatusEntry], NSError> {
+
+		var returnArray = [StatusEntry]()
+
+		// Do this because GIT_STATUS_OPTIONS_INIT is unavailable in swift
+		let pointer = UnsafeMutablePointer<git_status_options>.allocate(capacity: 1)
+		let optionsResult = git_status_init_options(pointer, UInt32(GIT_STATUS_OPTIONS_VERSION))
+		guard optionsResult == GIT_OK.rawValue else {
+			return Result.failure(NSError(gitError: optionsResult, pointOfFailure: "git_status_init_options"))
+		}
+		var options = pointer.move()
+		pointer.deallocate(capacity: 1)
+
+		var unsafeStatus: OpaquePointer? = nil
+		let statusResult = git_status_list_new(&unsafeStatus, self.pointer, &options)
+		guard statusResult == GIT_OK.rawValue, let unwrapStatusResult = unsafeStatus else {
+			return Result.failure(NSError(gitError: statusResult, pointOfFailure: "git_status_list_new"))
+		}
+
+		let count = git_status_list_entrycount(unwrapStatusResult)
+
+		for i in 0..<count {
+			let s = git_status_byindex(unwrapStatusResult, i)
+			if s?.pointee.status.rawValue == GIT_STATUS_CURRENT.rawValue {
+				continue
+			}
+			var status: Status? = nil
+
+			var headToIndex: DiffDelta? = nil
+			var indexToWorkDir: DiffDelta? = nil
+
+			if let statusValue = s?.pointee.status.rawValue {
+				status = Status(rawValue: statusValue)
+			}
+
+			if let htoi = s?.pointee.head_to_index {
+				headToIndex = DiffDelta(from: htoi.pointee)
+			}
+
+			if let itow = s?.pointee.index_to_workdir {
+				indexToWorkDir = DiffDelta(from: itow.pointee)
+			}
+
+			let statusEntry = StatusEntry(status: status, headToIndex: headToIndex, indexToWorkDir: indexToWorkDir)
```

Comment:
Can we make the `StatusEntry.init` take the `git_status_entry` and do this work?